test/simulator/moves/trickroom.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+var assert = require('assert');
+var battle;
+
+describe('Trick Room', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should cause slower Pokemon to move before faster Pokemon in a priority bracket', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Bronzong', ability: 'heatproof', moves: ['spore', 'trickroom']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Ninjask', ability: 'speedboost', moves: ['poisonjab', 'spore']}]);
+		battle.choose('p1', 'move 2');
+		battle.commitDecisions();
+		battle.choose('p2', 'move 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].status, '');
+		assert.strictEqual(battle.p2.active[0].status, 'slp');
+	});
+
+	it('should not allow Pokemon using a lower priority move to act before other Pokemon', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Bronzong', ability: 'heatproof', moves: ['spore', 'trickroom']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Ninjask', ability: 'speedboost', moves: ['poisonjab', 'protect']}]);
+		battle.choose('p1', 'move 2');
+		battle.commitDecisions();
+		battle.choose('p2', 'move 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].status, '');
+		assert.strictEqual(battle.p2.active[0].status, '');
+	});
+
+	it('should also affect the activation order for abilities and other non-move actions', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [
+			{species: 'Bronzong', ability: 'heatproof', moves: ['trickroom', 'explosion']},
+			{species: 'Hippowdon', ability: 'sandstream', moves: ['protect']}
+		]);
+		battle.join('p2', 'Guest 2', 1, [
+			{species: 'Ninjask', ability: 'speedboost', moves: ['shellsmash']},
+			{species: 'Ninetales', ability: 'drought', moves: ['protect']}
+		]);
+		battle.commitDecisions();
+		battle.choose('p1', 'move 2');
+		battle.commitDecisions();
+		battle.choose('p1', 'switch 2');
+		battle.choose('p2', 'switch 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].speciesid, 'hippowdon');
+		assert.strictEqual(battle.p2.active[0].speciesid, 'ninetales');
+		assert.strictEqual(battle.effectiveWeather(), 'sunnyday');
+	});
+
+	// The following two tests involve the Trick Room glitch, where turn order changes when a Pokemon goes to 1809 speed.
+	// In these tests, we manually set the speed using events, but the effects can be approximately replicated with the following:
+	// 1808: Deoxys-Speed with 31 Speed IVs, 224 Speed EVs, neutral nature, with a +6 Speed boost.
+	// 1809: Ninjask with 31 Speed IVs, 184 Speed EVs, neutral nature, with a +4 Speed boost and a Choice Scarf.
+	// over 1810: Either of the above Pokemon with their Speed raised further (more EVs, boosts, Tailwind, etc.)
```

Comment:
I forgot to reply, sorry.
I meant: feed this Ninjask to the simulator and check that what happens is the same as in cartridge.